doc/programming/language/packing.rst

```diff
@@ -0,0 +1,57 @@
+
+.. _packing:
+
+==========================
+Packing / Unpacking Values
+==========================
+
+A few of Spicy's atomic types support low-level conversion from, or
+into, a binary representation through two operators:
+
+- ``pack(VALUE, ARGS)`` turns a ``VALUE`` into a sequence of raw bytes
+  representing the value in binary form. ``ARGS`` specify what
+  encoding to use for the representation; they are type-specific (see
+  below for a list).  The ``pack`` operator returns a ``bytes``
+  instance containing the encoded data.
+
+  As an example, ``pack(uint16(513), spicy::ByteOrder::Network)``
+  returns ``\x02\x01``, which is 513 in network byte order.
+
+- ``unpack<TYPE>(DATA, ARGS)`` parses a value of type ``TYPE`` from a
+  binary representation ``DATA``. ``ARGS`` specify what encoding to
+  expect for ``DATA``. The ``unpack`` operator returns a 2-tuple
+  ``(VALUE, REMAINDER)`` where ``VALUE`` is the parsed value, and
+  ``REMAINDER`` is any bytes left over from ``DATA`` that weren't used
+  for parsing the value. If parsing fails, ``unpack`` throws an
+  ``InvalidValue`` exception.
+
+  As an example, ``unpack<uint16>(b"\x02\x01XYZ",
+  spicy::ByteOrder::Network)`` returns the tuple ``(513, b"XYZ")``.
+
+The following table summarizes the types that currently support
```

Comment:
I wonder whether we should also add `pack` (and for symmetry `unpack`) for `bytes`. The intended use case would be to change the byte order of existing `bytes` values.

Reply:
Not sure a notion of byte-order makes sense for just a raw sequence of bytes? "Byte order" needs some kind of more-than-1-byte semantical unit. `pack` is about transforming something into bytes, so I think a `pack` *for* bytes could really just be a no-op.